### PR TITLE
Add `depth_ndc_to_view_z` for cpu-side

### PR DESCRIPTION
**Pull request by DasLixou**
# Objective

I want to get the visual depth (after view proj matrix stuff) of the object beneath my cursor.
Even when having a write-back of the depth texture, you would still need to convert the NDC depth to a logical value.

## Solution

This is done on shader-side by [this function](https://github.com/bevyengine/bevy/blob/e6261b0f5f1124ffa67b8fe9a2d24a2047795192/crates/bevy_pbr/src/render/view_transformations.wgsl#L151), which I ported over to the cpu-side.

I also added `world_to_viewport_with_depth` to get a `Vec3` instead of `Vec2`.

---

If anyone knows a smarter solution to get the visual depth instead of going `screen -> viewport ray -> screen`, please let me know :>

### CODE REVIEW

File: crates/bevy_render/src/camera/camera.rs [hunk 1]
```diff
@@ -373,6 +373,39 @@ impl Camera {
...
+    /// Given a position in world space, use the camera to compute the viewport-space coordinates and depth.
+    ///
+    /// To get the coordinates in Normalized Device Coordinates, you should use
+    /// [`world_to_ndc`](Self::world_to_ndc).
+    ///
+    /// Returns `None` if any of these conditions occur:
+    /// - The computed coordinates are beyond the near or far plane
+    /// - The logical viewport size cannot be computed. See [`logical_viewport_size`](Camera::logical_viewport_size)
+    /// - The world coordinates cannot be mapped to the Normalized Device Coordinates. See [`world_to_ndc`](Camera::world_to_ndc)
+    ///     May also panic if `glam_assert` is enabled. See [`world_to_ndc`](Camera::world_to_ndc).
```
Comment (alice-i-cecile) [inline crates/bevy_render/src/camera/camera.rs:385]:
```suggestion
    /// May also panic if `glam_assert` is enabled. See [`world_to_ndc`](Camera::world_to_ndc).
```

File: crates/bevy_render/src/camera/camera.rs [hunk 2]
```diff
@@ -373,6 +373,39 @@ impl Camera {
         Some(viewport_position)
     }
 
+    /// Given a position in world space, use the camera to compute the viewport-space coordinates and depth.
+    ///
+    /// To get the coordinates in Normalized Device Coordinates, you should use
+    /// [`world_to_ndc`](Self::world_to_ndc).
+    ///
+    /// Returns `None` if any of these conditions occur:
```
Comment (alice-i-cecile) [inline crates/bevy_render/src/camera/camera.rs:381]: Non-blocking: this feels like a case where a `Result` would be clearer / more helpful for debugging: multiple failure modes feel like a custom error type is the right choice.

**Review by alice-i-cecile — APPROVED** [review]

File: crates/bevy_render/src/camera/camera.rs (hunk 1 above)
Comment (DasLixou) [reply crates/bevy_render/src/camera/camera.rs:385]: It's copied from the `world_to_viewport` function from above, should I also change it there?

File: crates/bevy_render/src/camera/camera.rs (hunk 2 above)
Comment (DasLixou) [reply crates/bevy_render/src/camera/camera.rs:381]: Same here: just an extended version of `world_to_viewport`.
Could do the same there, but maybe in a follow up PR?
That would also make it a breaking change.

Comment (alice-i-cecile) [reply crates/bevy_render/src/camera/camera.rs:381]: Yep, let's swap all of these over in a follow-up.

**Pull request merged**